atomic_reactor/inner.py

```diff
@@ -252,6 +254,52 @@ def all_registries(self):
         return self.docker_registries + self.pulp_registries
 
 
+class FSWatcher(threading.Thread):
+
+    def __init__(self, *args, **kwargs):
+        super(FSWatcher, self).__init__(*args, **kwargs)
+        self.daemon = True  # exits whenever the process exits
+        self._lock = threading.RLock()
+        self._data = {}
+
+    def run(self):
+        while True:
+            with self._lock:
+                self._update(self._data)
+            os.sleep(1)
+
+    def get_data(self):
+        with self._lock:
+            data_copy = self._data.copy()
+        return data_copy
+
+
+    @staticmethod
+    def _update(data):
+        try:
+            st = os.statvfs("/")
+        except Exception as e:
+            return e
+
+        mb = 1000 ** 2  # sadly storage is generally expressed in decimal units
+        new_data = dict(
+            mb_free=st.f_bavail * st.f_frsize / mb,
+            mb_total=st.f_blocks * st.f_frsize / mb,
+            mb_used=(st.f_blocks - st.f_bfree) * st.f_frsize / mb,
+            inodes_free=st.f_ffree,
+            inodes_total=st.f_files,
+            inodes_used=st.f_files - st.f_ffree,
+        )
+        for key in ["mb_total", "mb_used", "inodes_total", "inodes_used"]:
+            if new_data[key] > data.get(key):
```

Comment:
`dict.get` returns `None` if key is set. In python2 `None` is evaluated as greater than for any integer. In python 3 it raises `TypeError`. Does it make sense to use `0` as the default value (`data.get(key, 0)`)? Or is checking for falsy value better?

Reply:
Ah, did not know that, thanks. I think `data.get(key, 0)` is better, can use it the same here and below.

Reply:
Do PR tests run under both python 2 and 3?

Reply:
I think if you use it below, it's going to mess up the "min" check.
I'm wondering if you need `sys.maxsize` for that one.
*Note that this is max **int** size, you can still get a larger value but it'll be of type **long**.

Reply:
Yes, they run for both 2 and 3. It looks like the tests for python 3 failed: https://travis-ci.org/projectatomic/atomic-reactor/builds/367111697?utm_source=github_status&utm_medium=notification